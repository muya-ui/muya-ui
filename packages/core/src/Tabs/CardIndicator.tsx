import styled, { css } from 'styled-components';

import { IThemedBaseProps } from '../types';
import { withThemeForStyled } from '../utils/withTheme';
import { ITabsProps } from './types';

function indicatorCss(props: ITabsProps & IThemedBaseProps) {
  const { theme } = props;
  const { Tabs: token } = theme.components;
  const { indicator } = token;
  return css`
    position: absolute;
    width: 100%;
    bottom: 0;
    left: 0;
    height: ${indicator.bgHeight}px;
    background: ${indicator.bgColor};
  `;
}

const CardIndicator = styled.div`
  ${indicatorCss}
`;
export default withThemeForStyled(CardIndicator);
