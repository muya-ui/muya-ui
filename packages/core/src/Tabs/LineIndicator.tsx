import styled, { css } from 'styled-components';

import { IThemedBaseProps } from '../types';
import { withThemeForStyled } from '../utils/withTheme';

export interface ILineIndicatorProps {
  $width?: number;
  $left?: number;
  $transition?: 'on' | 'off';
  /**
   * 隐藏指示器
   */
  $hide?: boolean;
}

function tabIndicatorCss(props: ILineIndicatorProps & IThemedBaseProps) {
  const { theme, $left = 0, $width = 0, $transition = 'off', $hide = false } = props;
  const { Tabs: token } = theme.components;
  const { indicator } = token;
  const {
    pattern: { duration, easing },
  } = theme.transition;
  let transitionCss;
  if ($transition === 'on') {
    transitionCss = css`
      transition: all ${duration.status}ms ${easing.status};
    `;
  }
  let indicatorCss;
  if (!$hide) {
    indicatorCss = css`
      z-index: 2;
      &::before {
        content: '';
        display: block;
        position: absolute;
        bottom: 0;
        height: ${indicator.activeHeight}px;
        background: ${indicator.activeColor};
        width: ${$width}px;
        left: ${$left}px;
        ${transitionCss}
      }
    `;
  }
  return css`
    height: ${indicator.bgHeight}px;
    background: ${indicator.bgColor};
    ${indicatorCss}
  `;
}
const LineIndicator = styled.div<ILineIndicatorProps>`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  ${tabIndicatorCss}
`;

export default withThemeForStyled(LineIndicator);
