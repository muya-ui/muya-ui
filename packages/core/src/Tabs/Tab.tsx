import { ITheme } from '@muya-ui/theme-light';

import React, { useMemo } from 'react';

import styled, { css, ThemedStyledProps } from 'styled-components';

import { IInlineButtonProps, InlineButton } from '../Button';
import addPx from '../utils/addPx';
import memoForwardRef from '../utils/memoForwardRef';
import useStyles from '../utils/useStyles';
import { withThemeForStyled } from '../utils/withTheme';

import { ITabProps } from './types';

const InnerButton = styled(InlineButton)``;
const BaseNode = styled.div``;

const TabPure = memoForwardRef<HTMLDivElement, ITabProps>((props, ref) => {
  const {
    index,
    selected,
    type,
    divider,
    buttonType: buttonTypeProp,
    // InlineButton
    size,
    htmlType,
    disabled,
    busy,
    component: asProp,
    prefixNode,
    suffixNode,
    onClick,
    loading,
    textFine,
    fontWeight,
    width,
    constant,
    weakLevel,
    href,
    target,
    rel,
    showIndicator,

    styles,
    children,
    ...otherProps
  } = props;
  const innerStyles = useStyles<'item'>(
    'tab',
    {
      item: '',
    },
    styles,
  );

  const buttonType = useMemo(() => {
    if (buttonTypeProp) {
      return buttonTypeProp;
    }
    if (selected) {
      return 'primary';
    }
    if (type === 'card') {
      return 'normal';
    }
    return 'strong';
  }, [buttonTypeProp, selected, type]);

  const buttonProps: IInlineButtonProps = {
    type: buttonType,
    size,
    htmlType,
    disabled,
    onClick,
    busy,
    fontWeight: 'lighter',
    component: asProp,
    prefixNode,
    suffixNode,
    loading,
    width,
    constant,
    weakLevel,
    href,
    target,
    rel,
    styles,
    ...innerStyles.item,
  };

  return (
    <BaseNode {...otherProps} ref={ref}>
      <InnerButton {...buttonProps}>{children}</InnerButton>
    </BaseNode>
  );
});

function lineTabCss(props: ThemedStyledProps<ITabProps, ITheme>) {
  const { theme, size = 'm', selected, showIndicator = false } = props;
  const { Tabs: token } = theme.components;
  const height = token.height[size];
  const padding = token.padding[size];
  let paddingCss;
  if (padding) {
    paddingCss = css`
      padding: ${padding};
    `;
  }
  let selectedCss;
  if (selected && showIndicator) {
    selectedCss = css`
      &::before {
        content: '';
        display: block;
        position: absolute;
        bottom: 0;
        height: ${token.indicator.activeHeight}px;
        background: ${token.indicator.activeColor};
        width: 100%;
      }
    `;
  }
  return css`
    & ${InnerButton} {
      min-height: ${addPx(height)};
      ${paddingCss}
    }

    position: relative;
    margin-right: ${token.marginRight[size]}px;
    ${selectedCss}
  `;
}

function cardTabCss(props: ThemedStyledProps<ITabProps, ITheme>) {
  const { theme, size = 'm', selected, disabled, busy, divider = 'on' } = props;
  const { colors } = theme;
  const { card } = theme.components.Tabs;
  const height = card.height[size];
  const hasMargin = card.marginRight > 0;

  const interactive = !(busy || disabled);
  let statusCss;
  if (selected) {
    statusCss = css`
      border-color: ${colors.pattern.border.normal};
      border-bottom: none;
      border-top-left-radius: ${card.borderRadius};
      border-top-right-radius: ${card.borderRadius};
      background-color: ${card.bgSelectedColor};
      z-index: 1;
    `;
  } else {
    statusCss = css`
      background-color: ${card.bgColor};
      border-bottom-color: ${colors.pattern.border.normal};
      ${hasMargin &&
        css`
          border-color: ${colors.pattern.border.normal};
          border-bottom: none;
          border-top-left-radius: ${card.borderRadius};
          border-top-right-radius: ${card.borderRadius};
        `}
      ${interactive &&
        css`
          &:hover {
            background-color: ${card.bgHoverColor};
          }
        `}
    `;
  }
  let orderCss;
  const padding = card.padding[size];
  if (divider === 'on') {
    const dividerSize = card.dividerSize[size];
    orderCss = css`
      &::before {
        content: '';
        display: block;
        position: absolute;
        top: ${(height - dividerSize) / 2}px;
        right: -1px;
        height: ${dividerSize}px;
        width: 1px;
        background: ${colors.pattern.border.normal};
      }
    `;
  }
  return css`
    position: relative;
    border: 1px solid transparent;
    margin-right: ${card.marginRight}px;
    & ${InnerButton} {
      height: ${height}px;
      padding: 0 ${padding}px;
      line-height: 1;
    }

    ${statusCss}
    ${orderCss}
    &:last-child {
      border-top-right-radius: ${card.borderRadius};
      margin-right: 0;
    }
  `;
}

const Tab = styled(TabPure)`
  box-sizing: border-box;
  ${(props: ThemedStyledProps<ITabProps, ITheme>) =>
    props.type === 'card' ? cardTabCss(props) : lineTabCss(props)}
`;

export default withThemeForStyled(Tab);
