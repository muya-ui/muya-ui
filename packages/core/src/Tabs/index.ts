export { default } from './Tabs';
export { default as Tabs } from './Tabs';
export { default as Tab } from './Tab';
export * from './Tabs';
export * from './types';
