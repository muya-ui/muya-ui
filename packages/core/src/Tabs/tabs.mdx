---
name: Tabs 选项卡
menu: 组件
route: /components/tabs
autoDemos: true
parent: 导航
---

# Tabs 选项卡

用来分隔内容上有关联但属于不同类别的内容集合，并提供快速切换的能力。

## 使用场景

提供平级的区域将复杂的内容进行分组归纳与展示，让用户可以在不同子任务、视图、模式等之间切换。如：

- 你希望用户能在一个主页面内浏览多个相关子视图的内容。
- 你希望用户能在当前页面查看所有内容类型，并能方便地切换来查看更具体的内容类型。

{Demos}

### API

#### ITabsProps

<Props propsName="ITabsProps" />

#### ITabProps

`Tab` 可以使用 `InlineButton` 除了 `type` 外的所有属性。

<Props propsName="ITabProps" omit={['textFine']} />

### Styles

#### ITabsStyleKeys

<Styles typeName="ITabsStyleKeys" componentName="tabs" />

#### ITabStyleKeys

<Styles typeName="ITabStyleKeys" componentName="tab" />
