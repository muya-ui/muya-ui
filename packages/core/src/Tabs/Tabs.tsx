import React, { useRef } from 'react';
import styled, { css } from 'styled-components';

import { useForkRef } from '@muya-ui/utils';

import memoForwardRef from '../utils/memoForwardRef';
import useStyles from '../utils/useStyles';
import { withThemeForStyled } from '../utils/withTheme';
import CardIndicator from './CardIndicator';
import LineIndicator from './LineIndicator';
import TabsContainer from './TabsContainer';
import { ITabsProps } from './types';
import useLineTabs from './useLineTabs';
import useTabsBase from './useTabsBase';

const BaseNode = styled.div``;
const ExtraContentWrapper = styled.div``;

const defaultStyles = {
  container: '',
  indicator: '',
  extraContentWrapper: '',
};

const LineTabs = memoForwardRef<HTMLDivElement, ITabsProps>((props, ref) => {
  const {
    type,
    index,
    size,
    disabled,
    busy,
    onChange,
    styles,
    swipe,
    // 其余的
    children,
    height,
    tabBarExtraContent,
    lineIndicatorMode,

    // from swipe
    enableDiffChildren,
    equalNum,
    gutter,
    duration,

    ...otherProps
  } = props;
  const innerStyles = useStyles('tabs', defaultStyles, styles);
  const containerRef = useRef<HTMLDivElement>(null);
  const handleRef = useForkRef<HTMLDivElement>(containerRef, ref);
  const swipeProps = {
    enableDiffChildren,
    equalNum,
    gutter,
    duration,
  };
  const {
    children: resultChildren,
    indicatorState,
    onContainerChange,
    hideContainerIndicator,
  } = useLineTabs(props, containerRef);

  return (
    <BaseNode {...otherProps} ref={handleRef}>
      <TabsContainer
        {...innerStyles.container}
        {...swipeProps}
        size={size}
        type="line"
        swipe={swipe}
        height={height}
        onChange={onContainerChange}
      >
        {resultChildren}
      </TabsContainer>
      {tabBarExtraContent ? (
        <ExtraContentWrapper {...innerStyles.extraContentWrapper}>
          {tabBarExtraContent}
        </ExtraContentWrapper>
      ) : null}
      <LineIndicator
        {...innerStyles.indicator}
        $hide={hideContainerIndicator}
        $left={indicatorState.left}
        $width={indicatorState.width}
        $transition={indicatorState.transition}
      />
    </BaseNode>
  );
});
const CardTabs = memoForwardRef<HTMLDivElement, ITabsProps>((props, ref) => {
  const {
    type,
    index = 0,
    size,
    disabled,
    busy,
    onChange,
    styles,
    swipe,
    tabBarExtraContent,
    lineIndicatorMode,
    // 其余的
    children,

    // from swipe
    enableDiffChildren,
    equalNum,
    gutter,
    duration,
    ...otherProps
  } = props;
  const innerStyles = useStyles('tabs', defaultStyles, styles);
  const swipeProps = {
    enableDiffChildren,
    equalNum,
    gutter,
    duration,
  };
  const { children: resultChildren } = useTabsBase(props);
  return (
    <BaseNode {...otherProps} ref={ref}>
      <TabsContainer
        {...innerStyles.container}
        {...swipeProps}
        swipe={swipe}
        size={size}
        type="card"
      >
        {resultChildren}
      </TabsContainer>
      {tabBarExtraContent ? (
        <ExtraContentWrapper {...innerStyles.extraContentWrapper}>
          {tabBarExtraContent}
        </ExtraContentWrapper>
      ) : null}
      <CardIndicator {...innerStyles.indicator} />
    </BaseNode>
  );
});

const TabsPure = memoForwardRef<HTMLDivElement, ITabsProps>((props, ref) => {
  const { type = 'line', ...otherProps } = props;
  if (type === 'card') {
    return <CardTabs ref={ref} type="card" {...otherProps} />;
  }
  return <LineTabs ref={ref} type="line" {...otherProps} />;
});

const Tabs = styled(TabsPure)`
  position: relative;
  overflow: hidden;
  display: flex;
  align-items: center;
  ${props => {
    const { theme, size = 'm', type = 'line' } = props;
    const { Tabs: token } = theme.components;
    const padding = token.padding[size];
    let paddingCss = css`
      /* card indicator 下方背景线bottom: 0，会遮盖extraContent的内容，需要额外处理 */
      padding-bottom: ${token.indicator.bgHeight}px;
    `;
    // line indicator 需要为extraContent增加paddingBottom，确保内容对齐
    if (type === 'line' && padding) {
      paddingCss = css`
        padding: ${padding};
      `;
    }
    return css`
      & ${ExtraContentWrapper} {
        ${paddingCss};
      }
    `;
  }}
`;

export default withThemeForStyled(Tabs);
