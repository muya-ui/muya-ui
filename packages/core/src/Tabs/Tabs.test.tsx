import React from 'react';
import renderer from 'react-test-renderer';

import Tab from './Tab';
import Tabs from './Tabs';

test('测试 Tabs line no tab', () => {
  const tree = renderer.create(<Tabs></Tabs>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs line no valid tab', () => {
  const tree = renderer.create(<Tabs>ss</Tabs>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs line', () => {
  const tree = renderer
    .create(
      <Tabs>
        <Tab>1</Tab>
        <Tab>2</Tab>
        <Tab>3</Tab>
      </Tabs>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs 自定义高度，文字水平居中', () => {
  const tree = renderer
    .create(
      <Tabs height={60}>
        <Tab>1</Tab>
        <Tab>2</Tab>
        <Tab>3</Tab>
      </Tabs>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs card no tab', () => {
  const tree = renderer.create(<Tabs type="card"></Tabs>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs card no valid tab', () => {
  const tree = renderer.create(<Tabs type="card">ss</Tabs>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs card', () => {
  const tree = renderer
    .create(
      <Tabs type="card">
        <Tab>1</Tab>
        <Tab>2</Tab>
        <Tab>3</Tab>
      </Tabs>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 Tabs empty', () => {
  const tree = renderer.create(<Tabs type="card"></Tabs>).toJSON();
  expect(tree).toMatchSnapshot();
});
