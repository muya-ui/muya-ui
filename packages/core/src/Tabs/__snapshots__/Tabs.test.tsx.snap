// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Tabs card 1`] = `
.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(26,122,248,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(51,139,255,1);
}

.c5:active {
  color: rgba(12,108,235,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c10 + .c4 {
  margin-left: 8px;
}

.c7 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(102,102,102,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c7:hover {
  color: rgba(51,139,255,1);
}

.c7:active {
  color: rgba(12,108,235,1);
}

.c7 + .c4 {
  margin-left: 8px;
}

.c11 .c4 {
  min-height: 32px;
  padding: 0 0 8px 0;
}

.c2 {
  box-sizing: border-box;
  position: relative;
  border: 1px solid transparent;
  margin-right: 0px;
  border-color: rgba(238,238,238,1);
  border-bottom: none;
  border-top-left-radius: 2px;
  border-top-right-radius: 2px;
  background-color: rgba(255,255,255,1);
  z-index: 1;
}

.c2 .c4 {
  height: 30px;
  padding: 0 14px;
  line-height: 1;
}

.c2:last-child {
  border-top-right-radius: 2px;
  margin-right: 0;
}

.c6 {
  box-sizing: border-box;
  position: relative;
  border: 1px solid transparent;
  margin-right: 0px;
  background-color: rgba(248,248,248,1);
  border-bottom-color: rgba(238,238,238,1);
}

.c6 .c4 {
  height: 30px;
  padding: 0 14px;
  line-height: 1;
}

.c6:hover {
  background-color: rgba(242,242,242,1);
}

.c6::before {
  content: '';
  display: block;
  position: absolute;
  top: 8px;
  right: -1px;
  height: 14px;
  width: 1px;
  background: rgba(238,238,238,1);
}

.c6:last-child {
  border-top-right-radius: 2px;
  margin-right: 0;
}

.c8 {
  box-sizing: border-box;
  position: relative;
  border: 1px solid transparent;
  margin-right: 0px;
  background-color: rgba(248,248,248,1);
  border-bottom-color: rgba(238,238,238,1);
}

.c8 .c4 {
  height: 30px;
  padding: 0 14px;
  line-height: 1;
}

.c8:hover {
  background-color: rgba(242,242,242,1);
}

.c8:last-child {
  border-top-right-radius: 2px;
  margin-right: 0;
}

.c9 {
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background: rgba(238,238,238,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c12 {
  padding-bottom: 1px;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  >
    <div
      className="c2"
    >
      <button
        className="c3 c4 c5 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="c6"
    >
      <button
        className="c3 c4 c7 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="c8"
    >
      <button
        className="c3 c4 c7 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
  </div>
  <div
    className="c9 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs card no tab 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background: rgba(238,238,238,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c3 {
  padding-bottom: 1px;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  />
  <div
    className="c2 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs card no valid tab 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background: rgba(238,238,238,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c3 {
  padding-bottom: 1px;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  />
  <div
    className="c2 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs empty 1`] = `
.c2 {
  position: absolute;
  width: 100%;
  bottom: 0;
  left: 0;
  height: 1px;
  background: rgba(238,238,238,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c3 {
  padding-bottom: 1px;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  />
  <div
    className="c2 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs line 1`] = `
.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(26,122,248,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(51,139,255,1);
}

.c5:active {
  color: rgba(12,108,235,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c6 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(51,51,51,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c6:hover {
  color: rgba(51,139,255,1);
}

.c6:active {
  color: rgba(12,108,235,1);
}

.c6 + .c4 {
  margin-left: 8px;
}

.c2 {
  box-sizing: border-box;
  position: relative;
  margin-right: 32px;
}

.c2 .c4 {
  min-height: 32px;
  padding: 0 0 8px 0;
}

.c7 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  background: rgba(238,238,238,1);
  z-index: 2;
}

.c7::before {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  background: rgba(26,122,248,1);
  width: 0px;
  left: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c8 {
  padding: 0 0 8px 0;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  >
    <div
      className="c2"
    >
      <button
        className="c3 c4 c5 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="c2"
    >
      <button
        className="c3 c4 c6 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="c2"
    >
      <button
        className="c3 c4 c6 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
  </div>
  <div
    className="c7 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs line no tab 1`] = `
.c2 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  background: rgba(238,238,238,1);
  z-index: 2;
}

.c2::before {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  background: rgba(26,122,248,1);
  width: 0px;
  left: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c3 {
  padding: 0 0 8px 0;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  />
  <div
    className="c2 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs line no valid tab 1`] = `
.c2 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  background: rgba(238,238,238,1);
  z-index: 2;
}

.c2::before {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  background: rgba(26,122,248,1);
  width: 0px;
  left: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c3 {
  padding: 0 0 8px 0;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  />
  <div
    className="c2 muya-tabs-indicator"
  />
</div>
`;

exports[`测试 Tabs 自定义高度，文字水平居中 1`] = `
.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(26,122,248,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(51,139,255,1);
}

.c5:active {
  color: rgba(12,108,235,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c6 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(51,51,51,1);
  font-weight: 400;
  font-size: 14px;
  cursor: pointer;
}

.c6:hover {
  color: rgba(51,139,255,1);
}

.c6:active {
  color: rgba(12,108,235,1);
}

.c6 + .c4 {
  margin-left: 8px;
}

.c2 {
  box-sizing: border-box;
  position: relative;
  margin-right: 32px;
}

.c2 .c4 {
  min-height: 32px;
  padding: 0 0 8px 0;
}

.c7 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  background: rgba(238,238,238,1);
  z-index: 2;
}

.c7::before {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  background: rgba(26,122,248,1);
  width: 0px;
  left: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-x: hidden;
  z-index: 1;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .c8 {
  padding: 0 0 8px 0;
}

<div
  className="c0"
>
  <div
    className="c1 muya-tabs-container"
  >
    <div
      className="c2"
    >
      <button
        className="c3 c4 c5 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="c2"
    >
      <button
        className="c3 c4 c6 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="c2"
    >
      <button
        className="c3 c4 c6 muya-tab-item"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
  </div>
  <div
    className="c7 muya-tabs-indicator"
  />
</div>
`;
